cloudmarker/checks/firewallrulecheck.py

```diff
@@ -0,0 +1,119 @@
+        # to check for misconfiguration as GCP docs says that: The
+        # protocol type is required when creating a firewall rule. This
+        # value can either be one of the following well known protocol
+        # strings (tcp, udp, icmp, esp, ah, ipip, sctp), or the IP
+        # protocol number, implying it cannot be *.
+        self._misconfigurations = {
+            'sourceRanges': [
+                '0.0.0.0/0',
+                '*'
+            ],
+            'destinationRanges': [
+                '0.0.0.0/0',
+                '*'
+            ],
+        }
+
+    def _process_rule(self, record):
+        _misconfigurations = []
+
+        for misconfig_field in self._misconfigurations:
+            if misconfig_field in record:
+                for record_misconfig_field in record[misconfig_field]:
+                    if (record_misconfig_field in
+                            self._misconfigurations[misconfig_field]):
+                        _misconfigurations.append(misconfig_field
+                                                  + ':'
+                                                  + record_misconfig_field)
+        return {
+            'misconfigurations': _misconfigurations,
+            'id': record['id']
+        }
+
+    def eval(self, record):
+        """Evaluate Firewall rules for a given record.
+
+        Evaluate the weak security misconfigurations for each record and
+        collate each weakly identified record with its misconfigurations. This
+        function will only evaluate firewall_rule record.
+
+        For each such record it will create a dict object containing the weak
+        misconfigurations and return those records.
+
+        Arguments:
+            record (dict): Dictionary containing the rules definition.
+
+        Yields:
+            anomaly (dict): Dictionary representing anomaly.
+
+        The anomaly record yielded by this method would look like:
+
+        .. code:: python
+
+            {
+                "record_type": "firewall_rule_anomaly",
+                "id": ""
+                "rule": "https://www.googleapis.com/compute/v1/projects"
+                        "/foo/global/firewalls/default-allow-http",
+                "network":"https://www.googleapis.com/compute/v1/projects"
+                          "/foo/global/networks/default",
+                "misconfigurations": [
+                    "sourceRanges(s):0.0.0.0/0",
+                    "destinationRanges(s):0.0.0.0/0",
+                ]
+                "id": "7890789078907890"
+            }
+
+        """
+        # Process firewall rule only if the record type is 'firewall_rule' and
+        # the rule is not disabled.
+        if (record['record_type'] != 'firewall_rule'
+                or record['disabled']):
+            return
+
+        _anomaly_record = {
+            "record_type": "firewall_rule_anomaly",
+            "anomaly_type": "weak_firewall_rules",
```

Comment:
Please use single-quotes instead of double-quotes for single-line strings within Python code. This is the convention we are following in this project. Here is the PEP 8 recommendation on this: https://www.python.org/dev/peps/pep-0008/#string-quotes.